File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maximum=0;
        int left=0,right=height.size()-1;
        while(left<right){
            maximum=max(maximum,(right-left)*min(height[left],height[right]));

            if(height[left]<height[right]){
                left++;
            }else{
                right--;
            }
        }
        return maximum;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    static {
        for (int i = 0; i < 100; i++) {
            maxArea(new int[] {0, 0});
        }   
    }
    public static int maxArea(int[] height) {
        int maxArea=0;
        int left=0,right=height.length-1;
        while(left<right){
            int len=Math.min(height[left],height[right]);
            int width=right-left;
            maxArea=Math.max(maxArea,len*width);

            if(height[left]<height[right]){
                left++;
            }else{
                right--;
            }
        }
        return maxArea;
    }
}